libraries/stdlib/src/kotlin/random/Random.kt

```diff
@@ -330,7 +459,9 @@ internal fun Int.takeUpperBits(bitCount: Int): Int =
     this.ushr(32 - bitCount) and (-bitCount).shr(31)
 
 internal fun checkRangeBounds(origin: Int, bound: Int) = require(bound > origin) { boundsErrorMessage(origin, bound) }
+internal fun checkUIntRangeBounds(origin: UInt, bound: UInt) = require(bound > origin) { boundsErrorMessage(origin, bound) }
 internal fun checkRangeBounds(origin: Long, bound: Long) = require(bound > origin) { boundsErrorMessage(origin, bound) }
+internal fun checkULongRangeBounds(origin: ULong, bound: ULong) = require(bound > origin) { boundsErrorMessage(origin, bound) }
```

Comment:
Since the other variants are all called `checkRangeBounds`, I would prefer that name for the unsigned variants as well.

Or, because there are so many of them now, just replace all of them with the single

    internal fun <T : Comparable<T>> checkRangeBounds(origin: T, bound: T) = require(bound > origin) { "Random range is empty: [$origin, $bound)." }

Reply:
I wanted to do that, but it's not possible.

In the JVM, both checkRangeBounds(long, long) and checkRangeBounds(ULong, ULong) have the same signature. They'll both have the same name in JVM, and the compiler didn't allow me to do that

Reply:
I liked your approach more, I'll see if it works

Reply:
We're going to mangle top-level function names with unsigned types in their signatures to avoid this type of clashes. It would be possible to rename function after that mangling is turned on.